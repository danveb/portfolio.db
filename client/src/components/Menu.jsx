import { Link } from "react-router-dom";
import { links } from "../constants/links";
import { socials } from "../constants/socials";
import { resumeLink } from "../constants/resume";
import "../styles/Menu.scss"; 

export default function Menu({ menuOpen, setMenuOpen }) {
    return (
        <div className={"menu " + (menuOpen && "active")}>
            <ul>
                {links.map((link)=> (
                    <li key={link.id} onClick={()=> setMenuOpen(!menuOpen)}>
                        <Link href={link.url}>{link.text}</Link>
                    </li>
                ))}
                <li>
                    {resumeLink.map((link) => (
                        <div key={link.id}>
                            <Link href={link.url} rel="noopener noreferrer" target="_blank">resume</Link>
                        </div>
                    ))}
                </li>
            </ul>
            <ul>
                {socials.map((social) => (
                    <li key={social.id}>
                        <Link to={social.url} target={social.target} rel={social.rel}>
                            <img className="social" src={social.img} alt={social.alt} />
                        </Link>
                    </li>
                ))}
            </ul>
        </div>
    )
}
