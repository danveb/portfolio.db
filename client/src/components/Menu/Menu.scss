.menu {
    width: 250px; 
    height: 100vh; 
    background-color: var(--supernavy); 
    position: fixed; 
    top: 0; 
    right: -250px; 
    z-index: 2; 
    display: flex; 
    flex-direction: column; 
    justify-content: center; 
    align-items: center; 
    transition: all 0.5s ease-in-out; 

    &.active {
        right: 0; 
    }

    ul {
        margin: 0; 
        padding: 0; 
        list-style: none; 
        font-size: 26px; 
        font-weight: 100; 
        color: var(--superwhite); 
        width: 60%; 

        li {
            margin-bottom: 20px; 

            a {
                font-size: inherit; 
                color: inherit; 
                text-decoration: none; 
            }

            &:hover {
                font-weight: 500; 
            }
        }
    }
}

.social {
    svg {
        width: 30px; 
        height: 30px; 
    }
}