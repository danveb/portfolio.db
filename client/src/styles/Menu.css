.menu__nav {
  height: 100vh; 
  width: 100%; 
  background-color: #fff; 
  position: fixed; 
  top: 50px; 
  right: -100vw; 
  z-index: 2; 
  display: flex; 
  flex-direction: column; 
  visibility: hidden; 
  transition: all 0.4s ease-in-out; 
}

@media screen and (min-width: 768px) {
  .menu__nav.active {
    transform: translateY(-100%); 
  }
}

.menu__nav.active {
  right: 0; 
  transition: all 0.6s ease-in; 
  visibility: visible; 
  position: fixed; 
}

.menu {
  padding: 15px 0; 
}

.menu ul {
  /* height: 100%;  */
  display: flex; 
  flex-direction: column; 
  font-size: 22px; 
}

.menu li {
  padding: 13px 15px; 
}

.menu a {
  font-size: 14px; 
  color: var(--rich-black); 
  text-decoration: none; 
  display: flex; 
  justify-content: space-between; 
  width: 100%; 
}

.social {
  width: 25px; 
  height: 25px; 
  object-fit: contain; 
}