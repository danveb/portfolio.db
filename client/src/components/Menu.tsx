import { Link } from "react-router-dom";
import { links } from "../constants/links";
import { menuSocials } from "../constants/menu";
import { resumeLink } from "../constants/resume";
import "../styles/Menu.css"; 

import { openProps } from "../types";

export default function Menu({ menuOpen, setMenuOpen }: openProps) {
  return (
    <nav data-testid="menuId" className={"menu__nav " + (menuOpen && "active")}>
      <div className="menu">
        <ul>
          {links.map((link)=> (
            <li key={link.id} onClick={()=> setMenuOpen(!menuOpen)}>
              <Link to={link.url}>{link.text}</Link>
            </li>
          ))}
          <li>
            {resumeLink.map((link) => (
              <div key={link.id}>
                <Link to={link.url} rel="noopener noreferrer" target="_blank">resume</Link>
              </div>
            ))}
          </li>
        </ul>
        <ul>
          {menuSocials.map((social) => (
            <li key={social.id}>
              <Link to={social.url} target={social.target} rel={social.rel}>
                <img className="social" src={social.img} alt={social.alt} />
              </Link>
            </li>
          ))}
        </ul>
      </div>
    </nav>
  )
}
