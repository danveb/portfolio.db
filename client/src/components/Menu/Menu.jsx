import { links } from "../../constants/links";
import { socials } from "../../constants/socials";
import "./Menu.scss"; 

const Menu = ({ menuOpen, setMenuOpen }) => {
    return (
        <div className={"menu " + (menuOpen && "active")}>
            <ul>
                {links.map((link)=> (
                    <li key={link.id} onClick={()=> setMenuOpen(!menuOpen)}>
                        <a href={link.url}>{link.text}</a>
                    </li>
                ))}
                <li>
                    <a href="https://drive.google.com/file/d/12SyFoMp7IEL_Vkne9W3Akz5Nt0gB9YvE/view?usp=sharing" rel="noopener noreferrer" target="_blank">resume</a>
                </li>
            </ul>
            <ul>
                {socials.map((social) => (
                    <li key={social.id}>
                        <a href={social.url} target={social.target} rel={social.rel} className="social">{<social.icon />}</a>
                    </li>
                ))}
            </ul>
        </div>
    )
}

export default Menu