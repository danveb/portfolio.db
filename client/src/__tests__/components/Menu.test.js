import { render, screen } from "@testing-library/react"; 
import Menu from "../../components/Menu"; 
import { BrowserRouter } from "react-router-dom";

describe("Menu component", () => {
  // smoke test
  it("renders without crashing", () => {
    render(
      <BrowserRouter>
        <Menu />
      </BrowserRouter>
    ); 
  }); 

  // snapshot test
  test("matches snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Menu />
      </BrowserRouter>
    ); 
    expect(asFragment()).toMatchSnapshot(); 
  });

  // getByText
  test("should display the anchor tag", () => {
    render(
      <BrowserRouter>
        <Menu />
      </BrowserRouter>
    ); 
    const anchor = screen.getByText("resume"); 
    expect(anchor).toBeInTheDocument(); 
    expect(anchor).not.toHaveClass("world"); 
  });
});