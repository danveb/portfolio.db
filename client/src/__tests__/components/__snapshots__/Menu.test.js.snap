// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="menu__nav undefined"
  >
    <div
      class="menu"
    >
      <ul>
        <li>
          <a
            href="/about"
          >
            about
          </a>
        </li>
        <li>
          <a
            href="/projects"
          >
            projects
          </a>
        </li>
        <li>
          <a
            href="/contact"
          >
            contact
          </a>
        </li>
        <li>
          <div>
            <a
              href="https://drive.google.com/file/d/1N0VvKMKcAfogYAo-3gB3YONRT540EPLr/view?usp=sharing"
              rel="noopener noreferrer"
              target="_blank"
            >
              resume
            </a>
          </div>
        </li>
      </ul>
      <ul>
        <li>
          <a
            href="https://www.github.com/danveb"
            rel="noreferrer noopener"
            target="_blank"
          >
            <img
              alt="social icon"
              class="social"
              src="github.svg"
            />
          </a>
        </li>
        <li>
          <a
            href="https://www.linkedin.com/in/daniel-eze-bae"
            rel="noreferrer noopener"
            target="_blank"
          >
            <img
              alt="social icon"
              class="social"
              src="linkedin.svg"
            />
          </a>
        </li>
        <li>
          <a
            href="mailto:danibae91@gmail.com"
            rel="noreferrer noopener"
            target="_blank"
          >
            <img
              alt="social icon"
              class="social"
              src="at.svg"
            />
          </a>
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;
